buildSrc/src/main/java/org/mozilla/gradle/tasks/ValidateAndroidAppReleaseConfiguration.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.gradle.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import org.mozilla.gradle.GitAggregates
+import org.mozilla.gradle.ext.androidDSLOrThrow
+
+/**
+ * Validates that an Android app is correctly configured for release. When created, this
+ * task will automatically create dependencies upon the release tasks.
+ *
+ * This task currently asserts:
+ * - There are no uncommitted changes in git
+ * - The buildVersion in gradle matches the checked out git tag
+ *
+ * This task can be skipped by supplying the `-Pno-validate` argument.
+ */
+open class ValidateAndroidAppReleaseConfiguration : DefaultTask() {
+    init {
+        group = "Verification"
+        description = "Validates an Android app is correctly configured for release"
+
+        this.onlyIf { !project.hasProperty("no-validate") }
```

Comment:
Optional nit: I think the following would be more clear.  Might be worth a simple extension function.
```
project.doesntHaveProperty("no-validate")
```

Reply:
I'm not a fan of having a methods that wrap another method with a boolean state inversion for slightly more english readability because it bloats the number of method calls you can make on an object (reducing effectiveness of autocomplete) and requires additional maintenance overhead by violating the DRY principle (e.g. if a method is changed to support additional arguments).

If anything, I think this is on the language to make boolean inversion more readable (e.g. python's `if not condition:`). I guess you could make a function in kotlin such that, `if (not(condition())) {`...

As such, I'm going to keep it as is.